File: Vanilla/sources/zomboid/zombie/config/ConfigOption.java
package zombie.config;

public abstract class ConfigOption {
   protected final String name;

   public ConfigOption(String var1) {
      if (var1 != null && !var1.isEmpty() && !var1.contains("=")) {
         this.name = var1;
      } else {
         throw new IllegalArgumentException();
      }
   }

   public String getName() {
      return this.name;
   }

   public abstract String getType();

   public abstract void resetToDefault();

   public abstract void setDefaultToCurrentValue();

   public abstract void parse(String var1);

   public abstract String getValueAsString();

   public String getValueAsLuaString() {
      return this.getValueAsString();
   }

   public abstract void setValueFromObject(Object var1);

   public abstract Object getValueAsObject();

   public abstract boolean isValidString(String var1);
}

File: Vanilla/sources/zdoc/media/lua/shared/Library/ConfigOption.lua
---@class ConfigOption : zombie.config.ConfigOption
---@field protected name String
ConfigOption = {}

---@public
---@return String
function ConfigOption:getValueAsLuaString() end

---@public
---@param arg0 String
---@return boolean
function ConfigOption:isValidString(arg0) end

---@public
---@return String
function ConfigOption:getValueAsString() end

---@public
---@return String
function ConfigOption:getType() end

---@public
---@param arg0 Object
---@return void
function ConfigOption:setValueFromObject(arg0) end

---@public
---@return String
function ConfigOption:getName() end

---@public
---@return Object
function ConfigOption:getValueAsObject() end

---@public
---@return void
function ConfigOption:resetToDefault() end

---@public
---@param arg0 String
---@return void
function ConfigOption:parse(arg0) end

---@public
---@return void
function ConfigOption:setDefaultToCurrentValue() end
